t.in.eoarchive.py

```diff
@@ -0,0 +1,612 @@
+    if new_epsg != str(epsg):
+        grass.fatal(_("Creation of temporary location failed!"))
+
+    # return SRCGISRC, TMPLOC
+
+
+def get_actual_location():
+    """ Function that returns location and mapset of the current grass GISENV
+    Returns:
+        tgtloc (string): current location name
+        tgtmapset (string): current mapset name
+
+    """
+    global TGTGISRC, GISDBASE
+    # get actual location, mapset, ...
+    grassenv = grass.gisenv()
+    tgtloc = grassenv['LOCATION_NAME']
+    tgtmapset = grassenv['MAPSET']
+    GISDBASE = grassenv['GISDBASE']
+    TGTGISRC = os.environ['GISRC']
+    return tgtloc, tgtmapset
+
+
+def get_utmcells():
+    """ Function that queries the UTM Military Grid Reference System for Grid
+        tiles that overlap with the current computational region
+    Returns:
+        utm_tiles (list): list of UTM tiles that overlap with the current
+                          computational region
+
+    """
+    # save region
+    tmpregionname = 'tmp_region_utmgrid_' + str(os.getpid())
+    grass.run_command('v.in.region', output=tmpregionname, quiet=True)
+    rm_vectors.append(tmpregionname)
+
+    # get actual location, mapset, ...
+    tgtloc, tgtmapset = get_actual_location()
+
+    # create temporary location with epsg:4326
+    epsg = 4326
+    createTMPlocation(epsg)
+
+    # import grid
+    grass.run_command(
+        'v.proj', location=tgtloc, mapset=tgtmapset, input=tmpregionname,
+        output=tmpregionname, quiet=True)
+    grass.run_command('g.region', vector=tmpregionname, quiet=True)
+    grass.run_command(
+        'v.in.wfs',
+        url=options['wfs_mgrs'],
+        output='tmp_s2tiles_area', srs=epsg, name=options['wfs_name'],
+        flags='r', quiet=True)
+    utm_tiles = [x for x in grass.parse_command(
+        'v.db.select', map='tmp_s2tiles_area', column='name', flags='c')]
+
+    # switch to target location
+    os.environ['GISRC'] = str(TGTGISRC)
+
+    return utm_tiles
+
+
+def check_start_end(start, end):
+    """ Checks the validity of the user defined start and end date
+    Args:
+        start(string): user-defined start date in format YYYY-MM-DD
+        end(string): user-defined end date in format YYYY-MM-DD
+    Returns:
+        start_date(datetime): user-defined start date as datetime object
+        end_date(datetime): user-defined end date as datetime object
+
+    """
+    # set end if end is today
+    if end == "today":
+        end = date.today().strftime("%Y-%m-%d")
+    # check start after 2015-07
+    start_list = [int(x) for x in start.split('-')]
+    end_list = [int(x) for x in end.split('-')]
+    try:
+        start_date = date(start_list[0], start_list[1], start_list[2])
+        end_date = date(end_list[0], end_list[1], end_list[2])
+    except Exception as e:
+        grass.fatal(_(f"Start/End date is not defined in format YYYY-MM-DD: {e}"))
+    maja_date = date(2015, 7, 1)
+    if start_date < maja_date:
```

Comment:
I think it is ok if `start_date < maja_date`, what should not be allowed is `end_date < maja_date`

Reply:
This has been adapted